seissolxdmfwriter/seissolxdmfwriter/seissolxdmfwriter.py

```diff
@@ -0,0 +1,89 @@
+import h5py
+import numpy as np
+
+
+def write_seissol_xdmf(
+    prefix, nNodes, nCells, lDataName, lData, dt, node_per_element, lidt, reduce_precision
+):
+    precisionDict = {"int64": 8, "int32": 4, "float64": 8, "float32": 4}
+    numberTypeDict = {"int64": "UInt", "int32": "UInt", "float64": "Float", "float32": "Float"}
+    topology = "Tetrahedron" if node_per_element == 4 else "Triangle"
+    xdmf = """<?xml version="1.0" ?>
+<!DOCTYPE Xdmf SYSTEM "Xdmf.dtd" []>
+<Xdmf Version="2.0">
+ <Domain>"""
+    for i, idt in enumerate(lidt):
+        xdmf += f"""
+  <Grid Name="TimeSeries" GridType="Collection" CollectionType="Temporal">
+   <Grid Name="step_{idt}" GridType="Uniform">
+    <Topology TopologyType="{topology}" NumberOfElements="{nCells}">
+     <DataItem NumberType="Int" Precision="8" Format="HDF" Dimensions="{nCells} {node_per_element}">{prefix}.h5:/connect</DataItem>
+    </Topology>
+    <Geometry name="geo" GeometryType="XYZ" NumberOfElements="{nNodes}">
+     <DataItem NumberType="Float" Precision="8" Format="HDF" Dimensions="{nNodes} 3">{prefix}.h5:/geometry</DataItem>
+    </Geometry>
+    <Time Value="{idt*dt}"/>"""
+        for k, dataName in enumerate(lDataName):
+            prec = 4 if reduce_precision else precisionDict[lData[k].dtype.name]
+            number_type = numberTypeDict[lData[k].dtype.name]
+            xdmf += f"""
+    <Attribute Name="{dataName}" Center="Cell">
+     <DataItem ItemType="HyperSlab" Dimensions="{nCells}">
+      <DataItem NumberType="UInt" Precision="4" Format="XML" Dimensions="3 2">{i} 0 1 1 1 {nCells}</DataItem>
```

Comment:
Should be `NumberType="{number_type}" Precision="{prec}"`

Reply:
Here is an extract of a fault output:
``` xml
    <Attribute Name="SRs" Center="Cell">
     <DataItem ItemType="HyperSlab" Dimensions="6256">
      <DataItem NumberType="UInt" Precision="4" Format="XML" Dimensions="3 2">0 0 1 1 1 6256</DataItem>
      <DataItem NumberType="Float" Precision="8" Format="HDF" Dimensions="1 6256">tpv13-fault_cell.h5:/mesh0/SRs</DataItem>
     </DataItem>
    </Attribute>
```
So I think my code is consistent with seissol (while I don't understand why it is like that)